includes/core/classes/blocks/class-event-query.php

```diff
@@ -0,0 +1,324 @@
+<?php
+/**
+ * The "Event Query" class manages the core-block-variation,
+ * it mainly prepares the output of the block.
+ *
+ * @package GatherPress\Core
+ * @since 1.0.0
+ */
+
+namespace GatherPress\Core\Blocks;
+
+// Exit if accessed directly.
+defined( 'ABSPATH' ) || exit; // @codeCoverageIgnore
+
+use GatherPress\Core\Event;
+use GatherPress\Core\Traits\Singleton;
+use WP_Block;
+
+/**
+ * Class responsible for managing the "Event Query" block,
+ * which is a block-variation of 'core/query'.
+ *
+ * @since 1.0.0
+ */
+class Event_Query {
+	/**
+	 * Enforces a single instance of this class.
+	 */
+	use Singleton;
+
+	/**
+	 * Class constructor.
+	 *
+	 * This method initializes the object and sets up necessary hooks.
+	 *
+	 * @since 1.0.0
+	 */
+	protected function __construct() {
+		$this->setup_hooks();
+	}
+
+	/**
+	 * Set up hooks for various purposes.
+	 *
+	 * This method adds hooks for different purposes as needed.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return void
+	 */
+	protected function setup_hooks(): void {
+		add_action( 'init', array( $this, 'init' ), PHP_INT_MAX );
+	}
+
+	/**
+	 * Setup filters to handle rendering & REST requests for the block.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return void
+	 */
+	public function init(): void {
+
+		add_filter(
+			'pre_render_block',
+			array( $this, 'pre_render_block' ),
+			10,
+			2
+		);
+
+		// Updates the query vars for the Query Loop block in the block editor.
+		add_filter(
+			sprintf( 'rest_%s_query', Event::POST_TYPE ),
+			array( $this, 'rest_query' ),
+			10,
+			2
+		);
+
+		// We need more sortBy options.
+		add_filter(
+			sprintf( 'rest_%s_collection_params', Event::POST_TYPE ),
+			array( $this, 'rest_collection_params' )
+		);
+	}
+
+
+
+	/**
+	 * Allows render_block() to be short-circuited, by returning a non-null value.
+	 *
+	 * Updates the query on the front end based on custom query attributes.
+	 *
+	 * @param string|null $pre_render   The pre-rendered content. Default null.
+	 * @param array       $parsed_block The block being rendered.
+	 * @return string|null The pre-rendered content. Default null.
+	 */
+	public function pre_render_block( ?string $pre_render, array $parsed_block ): ?string {
+		if ( isset( $parsed_block['attrs']['namespace'] ) && 'gatherpress-event-query' === $parsed_block['attrs']['namespace'] ) {
+
+			// "Hijack the global query. It's a hack, but it works." Ryan Welcher
```

Comment:
Looks like this is a stray comment, you have it again below where you're overriding `$wp_query`